### ISSUE TITLE
Mono branch - qr pattern

### ISSUE THREAD

**Issue by zarmomin** [beltransen/velo2cam_calibration#16]
Hi,
I've been playing with the stereo calibration and that works great! I was wondering if you could tell me where the makers need to be placed in order to try the 'mono' version as well.
Thanks

**Comment by pirobot** [comment]
+1.  And what would you suggest to use as a good marker?

Thanks!

**Comment by pirobot** [comment]
Thank you @beltransen.  Yes we are very interested here at the Stanford Vision Lab.

**Comment by pirobot** [comment]
Hi @beltransen.  Just wondering if you are finished the paper yet? :)

**Comment by beltransen** [comment]
Hi, next week I'm starting a 4-months stay in another lab, so I've been really busy these months with the travel arrangements. My estimation for this work is to be finished by the end of April: we are currently during the experimentation phase, but it may be paused for a couple of weeks so that I can settle down in the new place. Afterwards, it shouldn't take more than two weeks to finish the remaining work. As we know you are waiting for the new version to be available, we will consider publishing the final code and instructions before finishing the writing :slightly_smiling_face:

**Comment by pirobot** [comment]
Hi @beltransen and many thanks for the update!  Yes, we are really keen to try your mono code since we have been desperately trying to calibrate our VLP-16's and Occam omni-directional camera that you can see on our [Jackrabbot research robot](http://svl.stanford.edu/projects/jackrabbot/).  We could even give you some feedback that you could add to your paper if that would help.  In the meantime, good luck with your move!

**Comment by akshayk91** [comment]
Any update on the mono branch? 
Thanks in advance!

**Comment by junzhang2016** [comment]
@beltransen @cguindel  Hi, Beltrán and Guindel, thank you for your nice work. May I ask do you publish the LiDAR-mono camera paper? Where can we read it? I am very interested.

**Comment by junzhang2016** [comment]
@pirobot Hi, Dear Patrick, may I ask have you solved the problem of LiDAR-mono camera calibration? If not, I can provide some advice also. My paper can be referenced. We achieved calibration btn a sparse 3D LiDAR (Velodyne VLP-16) and mono-camera (left camera of ZED) in this paper, using the method proposed by Geigers et al..

> J. Zhang, P. Siritanawan, Y. Yue, C. Yang, M. Wen and D. Wang, "A Two-step Method for Extrinsic Calibration between a Sparse 3D LiDAR and a Thermal Camera," 2018 15th International Conference on Control, Automation, Robotics and Vision (ICARCV), Singapore, 2018, pp. 1039-1044.
doi: 10.1109/ICARCV.2018.8581170

> https://www.researchgate.net/publication/329117026_A_Two-step_Method_for_Extrinsic_Calibration_between_a_Sparse_3D_LiDAR_and_a_Thermal_Camera

> https://ieeexplore.ieee.org/document/8581170



> A. Geiger, F. Moosmann, Ö. Car and B. Schuster, "Automatic camera and range sensor calibration using a single shot," 2012 IEEE International Conference on Robotics and Automation, Saint Paul, MN, 2012, pp. 3936-3943.
doi: 10.1109/ICRA.2012.6224570 
> http://www.cvlibs.net/publications/Geiger2012ICRA.pdf

**Comment by beltransen** [comment]
Hi all,
it's been a while since you asked for the LiDAR-monocular calibration tool, but it has just been released! 

It took us more time than expected, as we tested with different calibration targets, wanted to improve the overall performance of the method, and provide an easy-to-use software so that anyone with minimal ROS knowledge is able to use it. 
We hope that you will appreciate our effort when using this new version.

The new manuscript is already available in *arxiv*:

Beltrán, J., Guindel, C., and García, F. (2021). [Automatic Extrinsic Calibration Method for LiDAR and Camera Sensor Setups](https://arxiv.org/abs/2101.04431). arXiv:2101.04431 [cs.CV]. Submitted to IEEE Transactions on Intelligent Transportation Systems.
